src/Presentation/Nop.Web/Areas/Admin/Components/CommonStatistics.cs

```diff
@@ -47,25 +56,38 @@ public IViewComponentResult Invoke()
             if (_workContext.CurrentVendor != null)
                 return Content("");
 
-            var model = new CommonStatisticsModel
-            {
-                NumberOfOrders = _orderService.SearchOrders(
+            CommonStatisticsModel model = null;
+            if (!_commonSettings.LargeDatabase)
+                model = new CommonStatisticsModel
+                {
+                    NumberOfOrders = _orderService.SearchOrders(
                 pageIndex: 0,
                 pageSize: 1).TotalCount,
 
-                NumberOfCustomers = _customerService.GetAllCustomers(
+                    NumberOfCustomers = _customerService.GetAllCustomers(
                 customerRoleIds: new[] { _customerService.GetCustomerRoleBySystemName(SystemCustomerRoleNames.Registered).Id },
                 pageIndex: 0,
                 pageSize: 1).TotalCount,
 
-                NumberOfPendingReturnRequests = _returnRequestService.SearchReturnRequests(
+                    NumberOfPendingReturnRequests = _returnRequestService.SearchReturnRequests(
                 rs: ReturnRequestStatus.Pending,
                 pageIndex: 0,
                 pageSize: 1).TotalCount,
 
-                NumberOfLowStockProducts = _productService.GetLowStockProducts(0, 0, 1).TotalCount +
+                    NumberOfLowStockProducts = _productService.GetLowStockProducts(0, 0, 1).TotalCount +
                                              _productService.GetLowStockProductCombinations(0, 0, 1).TotalCount
-            };
+                };
+            else
+                model = new CommonStatisticsModel
+                {
+                    NumberOfOrders = _dbContext.Set<Order>().Where(x => !x.Deleted).Count(),
+
+                    NumberOfCustomers = _dbContext.SqlQuery<int>("select count(*) from Customer where Deleted = 0 and Id in (select Customer_Id from Customer_CustomerRole_Mapping where CustomerRole_Id = 3)").First(),
```

Comment:
This makes nopCommerce harder to maintain

Reply:
Xzelsius, is you are against LargeDatabase setting - propose something better.
About categories drop down:
1) If you have 20000 categories it takes forever to generate categories list for dropdown (because list contains full path for each category)
2) This list is very large and unusable.
3) It is transferred to client with each request. If client is mobile device then its terrible not only for bandwidth but also for cpu/ui usage.

_dbContext - used to execure SQL WHERE IN query.

If it makes harder to maintain - we can move sql queries to some one file like it is done with cache.

I think the goal is to make shop versatile and not easiest to maintain.
Easiest to maintain is one product shop. But nopcommerce with a LOT of its features is not the one by default.

Reply:
If there are 20 hardcoded SQL queries spread over 10 services and all for example are accessing Product and you for any reason change something related to that entity you probably have to change 20 SQL queries too. And thus have to remember where those queries are.

Instead of _dbContext.Set<TEntity> you should use IRepository<TEntity>.Table...

Regarding the dropdown problem with 20000 categories, how about a ajax powered search dropdown?
1000 times better than to know category ids for 20000 categories or to do a lookup anytime i want to map a category to a product.

Reply:
It's just a fast solution. Need to update my plugins first and I'm not sure these changes will be approved into nopcommerce. I don't need ajax search mostly because nearly all products are loaded/updated automatically.

Reply:
Instead of _dbContext.Set<TEntity> you should use IRepository<TEntity>.Table

Btw are there any real reasons that we should use IRepository instead of IDbContext?

Reply:
> Btw are there any real reasons that we should use IRepository instead of IDbContext?

**Except that nopCommerce does it this way**, i think the only reason will be plugins. A plugin could replace the repository for lets say UrlRecord and provide a custom implementation for storing them, an thus IDbContext could lead to an error.